packages/fast-usdc/src/utils/store.js

```diff
@@ -0,0 +1,114 @@
+import { Fail } from '@endo/errors';
+
+/**
+ * @import {Key} from '@endo/patterns';
+ */
+
+// TODO provide something like this in a more common place, perhaps as a BagStore
+/**
+ * Creates a bag (multi-set) API that wraps a MapStore where values are counts.
+ *
+ * @template {Key} K
+ * @param {MapStore<K, number>} mapStore
+ */
+export const asMultiset = mapStore =>
+  harden({
+    /**
+     * Add an item to the bag, incrementing its count.
+     *
+     * @param {K} item The item to add
+     * @param {number} [count] How many to add (defaults to 1)
+     */
+    add: (item, count = 1) => {
+      if (count <= 0) {
+        throw Fail`Cannot add a non-positive count ${count} to bag`;
+      }
+
+      if (mapStore.has(item)) {
+        const currentCount = mapStore.get(item);
+        mapStore.set(item, currentCount + count);
+      } else {
+        mapStore.init(item, count);
+      }
+    },
+
+    /**
+     * Remove an item from the bag, decrementing its count. If count reaches
+     * zero, the item is removed completely.
+     *
+     * @param {K} item The item to remove
+     * @param {number} [count] How many to remove (defaults to 1)
+     * @returns {boolean} Whether the removal was successful
+     */
+    remove: (item, count = 1) => {
+      if (count <= 0) {
+        throw Fail`Cannot remove a non-positive count ${count} from bag`;
+      }
+
+      if (!mapStore.has(item)) {
+        return false;
+      }
+
+      const currentCount = mapStore.get(item);
+      if (currentCount <= count) {
+        mapStore.delete(item);
```

Comment:
Feels like removing more than what is currently there should be an error.

Reply:
it's returning a success boolean. we can align on this when moved to another package.

Reply:
Asking to remove 5 from an entry that only has 3 still returns `true`, which seems incorrect.

Reply:
Revised: https://github.com/Agoric/agoric-sdk/compare/3e5eaec79bb4c246acefa5d279863cd0d98e0b5f..55979584e7e080cdf444f85a1228b1ce5f69074b